Formula/d/dqlite.rb

```diff
@@ -0,0 +1,161 @@
+class Dqlite < Formula
+  desc "Embeddable, replicated and fault-tolerant SQLite-powered engine"
+  homepage "https://dqlite.io"
+  url "https://github.com/canonical/dqlite/archive/refs/tags/v1.18.2.tar.gz"
+  sha256 "ce569a30958b6f673cbb7f8632706bc6868bc5b82ae4b04e0f63f3ad24125aaf"
+  license "LGPL-3.0-only"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "pkgconf" => :build
+  depends_on "go" => :test
+  depends_on "libuv"
+  depends_on :linux # requires Linux kernel AIO
+  depends_on "lz4"
+  depends_on "sqlite"
+
+  def install
+    system "autoreconf", "--force", "--install", "--verbose"
+    system "./configure", "--disable-silent-rules", *std_configure_args
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"testproject").mkpath
+    cd "testproject" do
+      (testpath/"testproject/main.go").write <<~EOS
```

Comment:
any upstream file to pull, seems a bit too big

Reply:
I trimmed it down from [this](https://github.com/canonical/go-dqlite/blob/master/cmd/dqlite-demo/dqlite-demo.go), which added two other Go dependencies that weren't strictly necessary for the test itself. I can switch over to that if desired.